### ISSUE TITLE
Sequence to Sequence LSTM prediction always gives same output

### ISSUE THREAD

**Issue by santoshaditham** [keras-team/keras#7964]
I have text docs for which I want to predict the titles. Each document is normalized to 200 words & each doc title to 5 words. So basically, show 200 words long doc & predict its 5 words long title. My dataset has 5000 docs & their titles.

First, I used word2vec and created word embeddings for the entire vocab [doc text & titles].

Then, I created the below model.

```
_activation = 'softmax'
_loss = 'categorical_crossentropy'
_optimizer = optimizers.RMSprop(lr=0.005, decay=1e-4)
_metrics = [metrics.categorical_accuracy]
model.add(LSTM(title_size*wordvec_size, batch_input_shape=(1, doc_words, wordvec_size), return_sequences=False, stateful=True, name="lstm_in"))
model.add(Dropout(0.25))
model.add(Reshape((title_size, wordvec_size), input_shape=(title_size*wordvec_size,)))
model.add(LSTM(wordvec_size, return_sequences=True, stateful=True, name="lstm_out"))
model.add(Dropout(0.25))
model.add(TimeDistributed(Dense(wordvec_size, activation=_activation)))
model.compile(loss=_loss, optimizer=_optimizer, metrics=_metrics)
for _ in range(10):
        model.fit(data_train, label_train, validation_split=1/10., batch_size=1, nb_epoch=2, verbose=0) #shuffle=True
        train_scores = model.evaluate(data_train, label_train, batch_size=1, verbose=0)
        test_scores = model.evaluate(data_test, label_test, batch_size=1, verbose=0)
        predicted_prob = model.predict(data_test, batch_size=1, verbose=0)
        #predicted_classes = model.predict_classes(data_test)
        model.reset_states()
```

But my model prediction is always the same even though the training accuracy, validation accuracy & testing accuracy is usually 35% or more. I am not sure if it is because the data is not good (not centered as mentioned in [this](https://github.com/fchollet/keras/issues/1727) issue) or if my model itself is wrong. Can someone help? Please let me know if you need more details.
